### DISCUSSION TITLE
Best practice with swr/tanstack-query?

### DISCUSSION THREAD

**Discussion by himself65** [pmndrs/zustand#2289] [Q&A] [answered]
Zustand is a local state management. But for the server data, we need refreshing and caching logic. That's why I need to use swr/tanstack. But I haven't find any discussion about this part

**Comment by dbritto-dev** [comment, marked as answer]
@himself65 you could use the`queryClient` directly in order to fetch data and store it on `zustand`

```ts
getUsers: async () => {
   set(() => ({ users: await queryClient.fetchQuery(...)  }))
}
```

**Reply by himself65** [reply]
Yes I could, but now the code is fallback to the no revalidate version. if I keep the page for hours, data will stale

**Reply by dbritto-dev** [reply]
@himself65 you could try subscribing to `queryCache` or `mutationCache` and then update the store

```ts
const queryCache = queryClient.getQueryCache()
queryCache.subscribe(() => { ... })
```

**Reply by dbritto-dev** [reply]
basically you should go through `react-query` and connect everything to `zustand`

**Reply by dbritto-dev** [reply]
may `jotai` is a better option in this case -> https://github.com/jotaijs/jotai-tanstack-query
